File: BayDialog.py

import sys
from PyQt4.QtCore import *
from PyQt4.QtGui import *

from BayDialog_ui import Ui_BayDialog





class BayDialog(QDialog):
    def __init__(self, Bay,  parent=None):
        QDialog.__init__(self, parent)
        self.ui = Ui_BayDialog()
        self.ui.setupUi(self)
        self.bay = Bay
        self.ui.positionSpin.setValue(self.bay.motor.getPosition())

        self.ui.materialCombo.clear()
        self.ui.materialCombo.addItem("")
        for material in self.bay.getMaterialList():
            self.ui.materialCombo.addItem(material)
            
        index= self.ui.materialCombo.findText(self.bay.getMaterial())
        if index!=-1: self.ui.materialCombo.setCurrentIndex(index)


        self.connect(self.ui.upButton, SIGNAL("released()"), self.volUp)
        self.connect(self.ui.downButton, SIGNAL("released()"), self.volDown)
        self.connect(self.ui.materialCombo,  SIGNAL("currentIndexChanged(int)"), self.updateBayMaterial)
        
    def volUp(self):
        currentCommand = self.ui.commandSpin.value()
        increment = self.ui.incrementSpin.value()
        self.ui.commandSpin.setValue(currentCommand-increment)
        self.ui.positionSpin.setValue(currentCommand-increment)
        
    def volDown(self):
        currentCommand = self.ui.commandSpin.value()
        increment = self.ui.incrementSpin.value()
        self.ui.commandSpin.setValue(currentCommand+increment)
        self.ui.positionSpin.setValue(currentCommand+increment)
    
    def updateBayMaterial(self):
        material = self.ui.materialCombo.currentText()
        self.bay.setMaterial(material)


if __name__ == "__main__":
    from TestBay import Bay
    app = QApplication(sys.argv)
    
    bay = Bay(0)
    bay.setMaterialList(("one", "two"))
    bay.setMaterial("two")
    myform = BayDialog(bay)
    myform.show()
    
    sys.exit(app.exec_())

File: baydialog.cpp
#include "baydialog.h"
#include "ui_baydialog.h"

BayDialog::BayDialog(QWidget *parent, QString bayName) :
    QDialog(parent),
    ui(new Ui::BayDialog)
{
    ui->setupUi(this);
    ui->bayLabel->setText(bayName);
    ui->incrementSpin->setValue(0.20);
    ui->acceptButton->setFocus();
    this->setWindowTitle("Edit Bay");
}

BayDialog::~BayDialog()
{
    delete ui;
}

void BayDialog::on_moveButton_clicked()
{
    QString bayName = ui->bayLabel->text().at(ui->bayLabel->text().length()-1);
    int bayNum = bayName.toInt();
    emit sendBayCommand(bayNum, ui->commandSpin->value(), true);
    ui->positionSpin->setValue(ui->commandSpin->value());
}

void BayDialog::on_acceptButton_clicked()
{
    this->close();
}

/*void BayDialog::on_upButton_released()
{
    int currentCommand = ui->commandSpin->value();
    int increment = ui->incrementSpin->value();
    ui->commandSpin->setValue(currentCommand-increment);
    ui->positionSpin->setValue(currentCommand-increment);
}

void BayDialog::on_downButton_released()
{
    int currentCommand = ui->commandSpin->value();
    int increment = ui->incrementSpin->value();
    ui->commandSpin->setValue(currentCommand+increment);
    ui->positionSpin->setValue(currentCommand+increment);
}*/

void BayDialog::on_upButton_clicked()
{
    double increment = ui->incrementSpin->value();
    double position = ui->positionSpin->value();
    QString bayName = ui->bayLabel->text().at(ui->bayLabel->text().length()-1);
    int bayNum = bayName.toInt();
    emit sendBayCommand(bayNum, -increment, false);
    ui->positionSpin->setValue(position-increment);
}

void BayDialog::on_downButton_clicked()
{
    double increment = ui->incrementSpin->value();
    double position = ui->positionSpin->value();
    QString bayName = ui->bayLabel->text().at(ui->bayLabel->text().length()-1);
    int bayNum = bayName.toInt();
    emit sendBayCommand(bayNum, +increment, false);
    ui->positionSpin->setValue(position+increment);
}

void BayDialog::on_commandSpin_editingFinished()
{
    this->setFocus();
}

void BayDialog::on_incrementSpin_editingFinished()
{
    this->setFocus();
}
